harness/determined/common/experimental/experiment.py

```diff
@@ -52,17 +52,31 @@ def __init__(
         self._id = experiment_id
         self._session = session
 
+        self.config = None  # type: Optional[Dict[str, Any]]
+
     @property
     def id(self) -> int:
         return self._id
 
-    def _get(self) -> bindings.v1Experiment:
+    def _get(self, session: Optional[api.Session] = None) -> bindings.v1Experiment:
```

Comment:
The existence of an optional session in methods for an object that is little more than a wrapper around id and session seems like a smell to me. Like maybe an `Experiment` should be defined another way. Could you please talk me through (either here in the comments or in a docstring, whichever you find appropriate) when a user (public or private) might want to override the session?

Reply:
At the moment, maybe it's specifically for the retries in `wait`. Per Ilia's earlier comment, maybe that logic can safely be used more generally.

Reply:
yeah, it's specifically for overriding retries. i also don't like that it's here, think i'm just going to follow ilia's suggestion.